pkg/controller/atlasproject/teams.go

```diff
@@ -0,0 +1,253 @@
+package atlasproject
+
+import (
+	"context"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/common"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/util/kube"
+
+	v1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+
+	controllerruntime "sigs.k8s.io/controller-runtime"
+
+	"go.mongodb.org/atlas/mongodbatlas"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/statushandler"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/watch"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/workflow"
+)
+
+type TeamDataContainer struct {
+	ProjectTeam *v1.Team
+	Team        *v1.AtlasTeam
+	Context     *workflow.Context
+}
+
+func (r *AtlasProjectReconciler) ensureAssignedTeams(ctx *workflow.Context, projectID string, project *v1.AtlasProject) workflow.Result {
+	resourcesToWatch := make([]watch.WatchedObject, 0, len(project.Spec.Teams))
+	defer func() {
+		r.EnsureMultiplesResourcesAreWatched(
+			types.NamespacedName{Namespace: project.Namespace, Name: project.Name},
+			r.Log, resourcesToWatch...,
+		)
+		r.Log.Debugf("watching team resources: %v\r\n", r.WatchedResources)
+	}()
+
+	teamsToAssign := map[string]*v1.Team{}
+	for _, assignedTeam := range project.Spec.Teams {
+		assignedTeam := assignedTeam
```

Comment:
I think that it's better to rename this var.

Reply:
@Sugar-pack do you think is really needed? the reassignment is only to avoid implicit memory aliasing

Reply:
Yeah. I think that it's better to avoid shadowing. But it's up to you.